apps/docs/src/components/TopBar/TopBar.svelte

```diff
@@ -1,13 +1,41 @@
 <script lang="ts">
-	import { ActionIcon, Tooltip, Menu, Divider, colorScheme } from '@svelteuidev/core';
-	import { Sun, Moon } from 'radix-icons-svelte';
+	import { ActionIcon, Tooltip, Menu, Divider, colorScheme, Modal, TextInput, Paper, Box, Kbd } from '@svelteuidev/core';
+	import { Sun, Moon, MagnifyingGlass } from 'radix-icons-svelte';
 	import { mobile } from 'components';
-	import { config } from './data';
+	import { config, searchLinks } from './data';
+  import { onMount } from 'svelte';
+  import { hotkey } from '@svelteuidev/composables';
+
+	let recentSearches,
+		searchTerm = '',
+		matchingSearches = [],
+		modalOpened = false;
 
 	function toggleTheme() {
 		colorScheme.update((v) => (v === 'light' ? 'dark' : 'light'));
 	}
 
+  function changeModalState() {
+		modalOpened = !modalOpened;
+    searchTerm = ""
+	}
+
+	onMount(() => recentSearches = JSON.parse(localStorage.getItem('recentSearches')) || []);
```

Comment:
For consistency, it would be better to be above all functions and before the variables, since it's the first thing called in the component